### ISSUE TITLE
Supabase onConflict not working when performing upsert

### ISSUE THREAD

**Issue by sonbs21** [GetDutchie/brick#465]
**Description**: When I perform an upsert on the `Heart` model with the `user_id` field marked as `onConflict`, I expect that the data will be updated if the `user_id` already exists. However, Supabase only performs an `INSERT` without the expected `UPDATE` when there is a conflict.

**Details**: I have configured the `onConflict` on `user_id` as follows:
```
@ConnectOfflineFirstWithSupabase(
  supabaseConfig: SupabaseSerializable(
    tableName: 'user', 
    onConflict: 'user_id'
  ),
  sqliteConfig: SqliteSerializable.defaults,
)
class Heart extends OfflineFirstWithSupabaseModel {
  @Supabase(unique: true, name: 'user_id')
  @Sqlite(unique: true)
  final String userId;

  final int heart;
  final DateTime? heartRecoveryTime;

  Heart({
    required this.userId,
    required this.heart,
    this.heartRecoveryTime,
  });
}
```
However, when performing the upsert, Supabase creates a new record instead of updating the existing one based on the `user_id`.

**Expected behavior**: I expect that when performing the upsert, if the `user_id` already exists, the record should be updated instead of creating a new one.

Please check and assist. Thank you!

**Comment by tshedor** [comment]
@sonbs21 Do you have a `UNIQUE INDEX` on this `user_id` column in Supabase? That should've prevented a duplicate record. Based on your model, you will also likely need to `@Sqlite(index: true, unique: true) @Supabase(unique:true)` your `heart` field.

**Comment by tshedor** [comment]
@sonbs21 Thank you for that PR. That's really, really helpful. Your code has been released in `brick_supabase: 1.0.5`
